pennylane/templates/subroutines/permute.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``permute`` template.
+"""
+
+import pennylane as qml
+
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.wires import Wires
+
+
+@template
+def Permute(permutation, wires):
+    r"""Applies a permutation to a set of wires.
+
+    Example usage:
+
+    .. code-block:: python
+
+        import pennylane as qml
+
+        with qml.tape.QuantumTape() as tape:
+            # Send contents of wire 4 to wire 0, of wire 2 to wire 1, etc.
+            qml.templates.Permute([4, 2, 0, 1, 3], wires=[0, 1, 2, 3, 4])
+
+    See "Usage Details" below for further examples.
+
+    Args:
+        permutation (list): A list of wire labels that represents the new ordering of wires
```

Comment:
It probably makes sense as is, but should we/could we support `Wires` as an input type?

Reply:
@mariaschuld would have the best insight here